### DISCUSSION TITLE
Why was `Optional` replaced with `Union[X, None]` in internal code?

### DISCUSSION THREAD

**Discussion by mattwelke** [fastapi/fastapi#5161] [Questions] [answered]
Still newish to Python, sorry if this is a n00b question. I saw this PR in the release notes for the latest version:

https://github.com/tiangolo/fastapi/pull/5124/files

I was wondering why one would make a change like this in Python? The Python docs at https://docs.python.org/3/library/typing.html#typing.Optional say `Optional` is shorthand. To me, it describes what's going on better. When I see `Union` I'd have to check what the valid options are, but the word optional immediately describes what's going on.

**Comment by zoliknemet** [comment, marked as answer]
Simple answer: tiangolo prefers Union
![image](https://user-images.githubusercontent.com/22326718/179209637-9fc3eff0-4ef2-4ec9-aac6-8cd8c4ded5f9.png)

**Reply by mattwelke** [reply]
lol makes sense. I guess in a way, it's more explicit. Maybe `Optional` sounds more like business logic than programming.

**Reply by fredrikaverpil** [reply]
You should mark the above answer as the accepted one if you are happy with this answer 😄 
This avoid lingering questions in the discussion forum.

**Reply by mattwelke** [reply]
Can't figure out how actually. I don't see an accept button or anything like that. Just an upvote button. Any idea how I can do that?

**Reply by fredrikaverpil** [reply]
Oh, my bad! 😢 Sorry for this noise!
TIL the category must be on "Question/Answer" format for this to work, like in the [Q&A](https://github.com/tiangolo/fastapi/discussions/categories/q-a?discussions_q=is%3Aanswered+category%3AQ%26A) one:

![](https://docs.github.com/assets/cb-69334/images/help/discussions/edit-category-details.png)

**Reply by tiangolo** [reply]
Yep, you can read more about it in the docs: https://fastapi.tiangolo.com/python-types/#using-union-or-optional
